sdcm/utils/remotewebbrowser.py

```diff
@@ -87,12 +88,20 @@ def browser(self):
     def get_screenshot(self, url, screenshot_path, resolution="1920px*1280px", load_page_screenshot_delay=30):
         LOGGER.info("Get a screenshot of %s", url)
 
-        self.browser.set_window_size(*resolution.replace("px", "").split("*", 1))
+        def wait_page_loading(_):
+            time.sleep(5)
```

Comment:
How this is different from the sleep you had before ? 

The whole idea of such function is to check something with the browser, if you don't use the we driver in this function, this is exactly as before, with more code

Reply:
We should look into using the formal service for saving snapshots and panels, it has all the nifty tricks that we are trying to get right:

https://github.com/grafana/grafana-image-renderer/blob/master/README.md

Reply:
This function is passed to web driver, which waiting timeout in remote browser context. This was simple solution to wait page load. 
More complex is to wait while each dashboard will be loaded. 
Each dashboard loaded in asynchronous mode, also the list of dashboards could change from version to version of monitor, in that case we need complex logic for selenium(seach dashboard, wait when elements in dashboard has been loaded etc). So i use simplest way
@bentsi may we need Trello task for more detailed solution?

Reply:
See 
https://github.com/scylladb/scylla-monitoring/issues/904#issuecomment-618704752

I strongly think we shouldn't go into this complexity when there's a formal solution to taking those screenshots we could build upon.

Reply:
This will be nice hack )

Reply:
It's not a hack if it's the formal was to capture snapshot out of grafana

Reply:
@fruch I think Alex tried to the Graphana image renderer ad part of the task https://trello.com/c/RHaPKcxZ/1462-fix-grafana-screenshots-monitor-31
but his conclusion was that it didn't work correctly, I remember it was stuck when taking a screenshot of the dashboard and worked well for the specific panel only.
That's why we decided not to use it. Maybe in new versions, they fixed the issues that they had, and we should check it again.
@aleksbykov Israel is right it is pointless to use WebDriverWait if the expected condition function does not verify anything in the DOM (like the appearance of some element). Read about it here: https://selenium-python.readthedocs.io/waits.html
So whether you remove it and add a much bigger timeout (that is a very quick-and-dirty and not robust solution) or you find a way to wait for availability/visibility of some element that indicates that all the panels are rendered and visible.

Reply:
I suggest for now, as a quick-and-dirty solution just increase the sleep time that we had to 120 sec an open a task for improvement. We need screenshots to work, they are broken too long

Reply:
@bentsi @fruch Added timeout for 150 seconds. Remove pointless WebDriverWait object for now.
@bentsi didn't find task in Trello. Create new on https://trello.com/c/OMPRDfd4